webapp/portlet/src/main/webapp/vue-apps/common/js/SiteService.js

```diff
@@ -41,12 +41,19 @@ export function getSites(siteType, excludedSiteType, excludedSiteName, excludeEm
   });
 }
 
-export function getSiteById(siteId, expandNavigations, excludeEmptyNavigationSites, lang) {
+export function getSiteById(siteId, expandNavigations, excludeEmptyNavigationSites, lang, visibility, excludeGroupNodesWithoutPageChildNodes, temporalCheck) {
```

Comment:
When the number of parameters becomes more than 3, it's hard to make the method maintainable. Please change the parameters in an object to be able to pass all query params.

```suggestion
export function getSiteById(siteId, params) {
```